src/js/letters/containers/AddressSection.jsx

```diff
@@ -8,26 +8,63 @@ import {
   isDomesticAddress,
   isMilitaryAddress,
   isInternationalAddress,
-  invalidAddressProperty,
   addressUpdateUnavailable
 } from '../utils/helpers.jsx';
-import { updateAddress, saveAddress } from '../actions/letters';
+import { saveAddress } from '../actions/letters';
 import Address from '../components/Address';
+import InvalidAddress from '../components/InvalidAddress';
+import AddressContent from '../components/AddressContent';
 
 export class AddressSection extends React.Component {
   constructor() {
     super();
-    this.state = { isEditingAddress: false };
+    this.state = {
+      hasLoadedAddress: false,
+      isEditingAddress: false,
+      editableAddress: {},
+    };
   }
 
-  handleUpdate = () => {
+  componentWillReceiveProps(nextProps) {
+    if (!this.state.hasLoadedAddress && Object.keys(nextProps.savedAddress).length > 0) {
+      this.setState({ hasLoadedAddress: true, editableAddress: nextProps.savedAddress });
+    }
+  }
+
+  handleSave = () => {
     this.setState({ isEditingAddress: false });
 
-    this.props.saveAddress(this.props.address);
+    this.props.saveAddress(this.state.editableAddress);
+  }
+
+  handleChange = (path, update) => {
+    this.setState(({ editableAddress }) => {
+      // reset state code when user changes address country but don't add or
+      // modify state property otherwise
+      return (path === 'country'
+        ? {
+          editableAddress: {
+            ...editableAddress,
+            [path]: update,
+            state: '',
+            militaryStateCode: '',
+          },
+        }
+        : {
+          editableAddress: {
+            ...editableAddress,
+            [path]: update,
+          },
+        }
+      );
+    });
   }
 
   render() {
-    const address = this.props.address || {};
+    // We want to use the Redux address as source of truth. Address in
+    // container state is only used to control the form input for the
+    // <Address/> component
```

Comment:
For clarification, you mean like if the save fails, we want to use / display the original address, so we can't override what's in the redux store.

Can we update this comment to include that? It makes perfect sense to me now, but I don't know that it would in six months is all.

Reply:
👍